app/components/MazemapEmbed/index.js

```diff
@@ -32,10 +35,60 @@ export const MazemapEmbed = ({ mazemapPoi, ...props }: Props) => {
       minZoom: 10,
       maxZoom: 20,
       zLevelControl: true,
-      scrollZoom: true,
+      scrollZoom: false,
       doubleClickZoom: false,
+      dragRotate: false,
+      dragPan: false,
       touchZoomRotate: false,
+      touchPitch: false, //this is a horrible feature
+      pitchWithRotate: false,
     });
+    embeddedMazemap.dragPan._mousePan.enable();
+
+    window.addEventListener('keydown', (e) => {
+      if (e.key === 'Control') {
+        controlPressed = true;
+      }
+    });
+    window.addEventListener('keyup', (e) => {
+      if (e.key === 'Control') {
+        controlPressed = false;
+      }
+    });
+
+    embeddedMazemap.on('wheel', () => {
+      if (controlPressed) {
+        embeddedMazemap.scrollZoom.enable();
+      } else {
+        embeddedMazemap.scrollZoom.disable();
+        setBlockScrollZoom(true);
+        setTimeout(() => {
+          setBlockScrollZoom(false);
+        }, 500);
```

Comment:
When scrolling over the map, the overlay will sometimes flicker a bit. This is because the timeout will set `setBlockScrollZoom(false);` even if we are still scrolling. To fix it you should save the timeout ID of the timeouts created, so you can cancel the previous timeout (if any) on each wheel event and start a new one. Something like this:
```javascript
if (blockScrollZoomTimeout) {
  clearTimeout(blockScrollZoomTimeout);
}
blockScrollZoomTimeout = setTimeout(() => {
  setBlockScrollZoom(false);
}, 500);
```

Reply:
Clever 👍